### PR TITLE
Go Lang upgarde from 1.17 to 1.18.5

### PR DESCRIPTION
**Pull request by imakowski**
Everything compiles fine

### CODE REVIEW

**Comment by millken** [comment]
Before upgrading, you need to ensure that ci can pass under 1.19

**Comment by dustinxie** [comment]
maybe let's upgrade to 1.18 first, 1.19 is too fresh a release

**Comment by raullenchai** [comment]
> maybe let's upgrade to 1.18 first, 1.19 is too fresh a release

I agree

**Comment by imakowski** [comment]
@dustinxie Do you know why tests are failing?

**Comment by CoderZhi** [comment]
```
2022/08/29 19:53:32 internal error: package "encoding/binary" without types was imported from "github.com/iotexproject/iotex-core/pkg/enc"
Error: Process completed with exit code 1.
```

I don't know the exact problem, but try `go mod tidy` first. Also, try `make all` locally first to make sure there is no problem from your client, and you can catch the problem earlier.

File: Dockerfile [hunk 1]
```diff
@@ -12,7 +12,7 @@ RUN go mod download
 COPY . .
 
 RUN mkdir -p $GOPATH/pkg/linux_amd64/github.com/iotexproject/ && \
-    make clean build-all
```
Comment (dustinxie) [inline Dockerfile:15]: revert this
`all` includes `test`, which is not necessary, and may fail in docker environment

**Comment by imakowski** [comment]
@dustinxie @CoderZhi 
Everything works fine on my machine 
ok      github.com/iotexproject/iotex-core/pkg/counter  5.607s
?       github.com/iotexproject/iotex-core/pkg/enc      [no test files]

I can not run `go.test.sh` because I don't have courtney but `make test `is passing

**Review by raullenchai — APPROVED** [review]

**Comment by raullenchai** [comment]
Pls fix the CI @imakowski

**Comment by imakowski** [comment]
@raullenchai problem is with self-hosted CI, can I have access to it?

**Comment by imakowski** [comment]
@raullenchai upgraded go-ethereum v1.10.4 => v1.10.23

**Comment by imakowski** [comment]
@dustinxie @CoderZhi can you help me on that?

**Comment by imakowski** [comment]
@Liuhaai @huof6829 Maybe you guys can help me on that?

**Comment by huof6829** [comment]
> @Liuhaai @huof6829 Maybe you guys can help me on that?

@imakowski   The version of `courtney` is old, you need update it.
```
go install github.com/dave/courtney@latest
```
I run successfully in local. My go version is 1.18.2 mac.

**Comment by imakowski** [comment]
@raullenchai @Liuhaai CI is passing but some tests are failing

**Comment by imakowski** [comment]
Can somebody merge this?

File: go.mod [hunk 2]
```diff
@@ -1,15 +1,14 @@
 module github.com/iotexproject/iotex-core
 
-go 1.17
+go 1.18
 
 require (
 	github.com/cenkalti/backoff v2.2.1+incompatible
-	github.com/ethereum/go-ethereum v1.10.4
```
Comment (dustinxie) [inline go.mod:7]: can you revert this? i don't think it's necessary?

**Review by CoderZhi — APPROVED** [review]

**Pull request merged**